rotkehlchen/db/settings.py

```diff
@@ -394,8 +404,15 @@ def reset(self) -> None:
     def update_entry(self, attr: str, value: DBSettingsFieldTypes) -> None:
         setattr(self._settings, attr, value)
 
-    def update_entries(self, settings: ModifiableDBSettings) -> None:
-        for attr in settings._fields:
+    def update_entries(self, settings: ModifiableDBSettings | DBSettings) -> None:
+
+        setting_fields: tuple[str, ...] | list[str]
+        if isinstance(settings, ModifiableDBSettings):
+            setting_fields = settings._fields
+        else:
+            setting_fields = [field.name for field in fields(settings)]
+
+        for attr in setting_fields:
```

Comment:
don't leave the empty line. Also why not make both a tuple?
```suggestion
    def update_entries(self, settings: ModifiableDBSettings | DBSettings) -> None:
        if isinstance(settings, ModifiableDBSettings):
            setting_fields = settings._fields
        else:
            setting_fields = tuple(field.name for field in fields(settings))

        for attr in setting_fields:
```